cli/src/commands/parse.rs

```diff
@@ -357,4 +413,65 @@ mod tests {
 
         assert_eq!(path, Some(sibling_lockfile_path));
     }
+
+    #[test]
+    fn test_relative_path() {
+        // Set up a temporary directory for testing.
+        let tempdir = tempfile::tempdir().unwrap();
+        let tempdir = tempdir.path().canonicalize().unwrap();
+        let curr_dir = env::current_dir().unwrap();
+
+        // Create a parent directory named "projects" within the temp directory.
+        let all_projects = tempdir.join("projects");
+        fs::create_dir_all(&all_projects).unwrap();
+
+        // Create a sample project directory named "sample" inside the "projects"
+        // directory. Also create a "Cargo.lock" file inside the "sample"
+        // directory.
+        let sample_dir = all_projects.join("sample");
+        let lockfile_path = sample_dir.join("Cargo.lock");
+        fs::create_dir_all(&sample_dir).unwrap();
+        File::create(&lockfile_path).unwrap();
+
+        // Change the current directory to the "sample" project directory.
+        env::set_current_dir(&sample_dir).unwrap();
```

Comment:
I'm a bit nervous about changing the current working directory inside of a unit test. These tests run in parallel, so if there is any other test that relies on the current working directory, this could cause issues...

I wonder if it would be beneficial to refactor a bit so that we can test a function that doesn't rely on the current working directory.